### PR TITLE
Improve OSX bit rot

### PR DESCRIPTION
**Pull request by michicc**
OSX APIs are an ever-moving target. Update some uses in OTTD to hopefully improve compatibility with newer OSX versions.

### CODE REVIEW

**Comment by andythenorth** [comment]
Compiles and runs for me on macOS 10.12.6.  I'm not competent to review the changes.  I could try and trigger specific issues that the commits address if that helps.

File: src/fontdetection.cpp [hunk 1]
```diff
@@ -381,39 +381,68 @@ FT_Error GetFontByFaceName(const char *font_name, FT_Face *face)
...
 
-	/* Get font reference from name. */
-	CFStringRef name = CFStringCreateWithCString(kCFAllocatorDefault, font_name, kCFStringEncodingUTF8);
-	ATSFontRef font = ATSFontFindFromName(name, kATSOptionFlagsDefault);
-	CFRelease(name);
-	if (font == kInvalidFont) return err;
-
-	/* Get a file system reference for the font. */
-	FSRef ref;
+    /* Get font reference from name. */
```
Comment (frosch123) [inline src/fontdetection.cpp]: indentation

File: src/fontdetection.cpp [hunk 2]
```diff
@@ -381,39 +381,68 @@ FT_Error GetFontByFaceName(const char *font_name, FT_Face *face)
...
+	UInt8 file_path[PATH_MAX];
 	OSStatus os_err = -1;
-#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5)
-	if (MacOSVersionIsAtLeast(10, 5, 0)) {
-		os_err = ATSFontGetFileReference(font, &ref);
-	} else
+    CFStringRef name = CFStringCreateWithCString(kCFAllocatorDefault, font_name, kCFStringEncodingUTF8);
+
+#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_6)
+    if (MacOSVersionIsAtLeast(10, 6, 0)) {
```
Comment (frosch123) [inline src/fontdetection.cpp]: indent

File: src/fontdetection.cpp [hunk 3]
```diff
@@ -381,39 +381,68 @@ FT_Error GetFontByFaceName(const char *font_name, FT_Face *face)
...
+		/* Loop over all matches until we can get a path for one of them. */
+		for (CFIndex i = 0; descs != NULL && i < CFArrayGetCount(descs) && os_err != noErr; i++) {
+			CTFontRef font = CTFontCreateWithFontDescriptor((CTFontDescriptorRef)CFArrayGetValueAtIndex(descs, i), 0.0, NULL);
+			CFURLRef fontURL = (CFURLRef)CTFontCopyAttribute(font, kCTFontURLAttribute);
+			if (CFURLGetFileSystemRepresentation(fontURL, true, file_path, lengthof(file_path))) os_err = noErr;
+			CFRelease(font);
+			CFRelease(fontURL);
+		}
+		if (descs != NULL) CFRelease(descs);
+    } else
```
Comment (frosch123) [inline src/fontdetection.cpp]: indent

**Comment by michicc** [comment]
Now with less spaces :)

**Comment by michicc** [comment]
Exterminate all the spaces!

**Review by TrueBrain — APPROVED** [review]
Like any of us has a clue; sounds good to me! :D

Tnx for taking care of Mac OS X :)

**Pull request merged**